sources/packages/api/src/route-controllers/education-program-offering/education-program-offering.controller.ts

```diff
@@ -257,20 +259,25 @@ export class EducationProgramOfferingController extends BaseController {
   @AllowAuthorizedParty(AuthorizedParties.institution)
   @HasLocationAccess("locationId")
   @Get(
-    "location/:locationId/education-program/:programId/program-year/:programYearId/offerings-list",
+    "location/:locationId/education-program/:programId/program-year/:programYearId/applicationId/:applicationId/offerings-list",
   )
   async getProgramOfferingsForLocationForInstitution(
     @Param("locationId") locationId: number,
     @Param("programId") programId: number,
     @Param("programYearId") programYearId: number,
+    @Param("applicationId") applicationId: number,
     @Query("includeInActivePY") includeInActivePY = false,
   ): Promise<OptionItem[]> {
+    const selectedOffering = (
```

Comment:
Is it not possible to receive directly the program intensity as a filter instead of the application id?

Reply:
The idea provide in the example (`?intensity=<Full Time/Part time>`) was to have the intensity in the query string to allow the same endpoint to be called with or without it, what means that the intensity would be acting as a filter instead of being part ot the URL. Is there a reason why the way that was implemented would be better?
@ann-aot @dheepak-aot @guru-aot, thoughts?

Reply:
No reason, I will refactor.

Reply:
Thanks for doing this change. One missing point is that the query string parameter is optional, so the validation and the DB query should take it into the consideration.